Update and rescale CFG denoising step

hdon96 opened the pull request:
Update as described by:
https://twitter.com/jeremyphoward/status/1584667278733324288 https://twitter.com/jeremyphoward/status/1585009792787304448

This requires using X/Y plot and small CFG scale (i.e. 0.15 work well) until the GUI slider parameters are modified. The new CFG update options could also be variously toggled with a GUI option.

hdon96 committed:
Update and rescale CFG denoising step

Update as described by:
https://twitter.com/jeremyphoward/status/1584667278733324288
https://twitter.com/jeremyphoward/status/1585009792787304448

This requires using X/Y plot and small CFG scale (i.e. 0.15 work well) until the GUI slider parameters are modified.
The new CFG update options could also be variously toggled with a GUI option.

hdon96 committed:
fix

fix it up. normal cfg ranges (1-15) seem to work fine

hdon96 committed:
clean up

clean up

hdon96 committed:
Update sd_samplers.py

that was a mistake...

hdon96 committed:
Update sd_samplers.py

forgot something

d8ahazard commented:
So, I edited my UI settings to adjust the values (allow for CFG scale of 0.15). Output images were...not great?

Looking at the repo of the fellow who proposed this, I see he's only changing one line in the DDIM/PLMS samplers.

https://github.com/CompVis/stable-diffusion/compare/main...sebderhy:stable-diffusion:rescale_guidance_weighted_avg#diff-5dca97583d86776b0adea0fff05029c70383da14fba1eb729cd69e52f5f6f05c

Just curious how you came to determine your modifications needed to go where they wound up.

Also - instead of changing the UI and breaking existing seeds, I'd recommend doing some kind of scaling on the CFG scale if "CFGv2" is enabled so that 7 roughly equates to 0.15 or whatever is worked out to be the magic number...so like, for 7, you'd divide it by 46.6666repeating.

hdon1 commented:
> So, I edited my UI settings to adjust the values (allow for CFG scale of 0.15). Output images were...not great?
> 
> Looking at the repo of the fellow who proposed this, I see he's only changing one line in the DDIM/PLMS samplers.
> 
> [CompVis/stable-diffusion@main...sebderhy:stable-diffusion:rescale_guidance_weighted_avg#diff-5dca97583d86776b0adea0fff05029c70383da14fba1eb729cd69e52f5f6f05c](https://github.com/CompVis/stable-diffusion/compare/main...sebderhy:stable-diffusion:rescale_guidance_weighted_avg#diff-5dca97583d86776b0adea0fff05029c70383da14fba1eb729cd69e52f5f6f05c)
> 
> Just curious how you came to determine your modifications needed to go where they wound up.
> 
> Also - instead of changing the UI and breaking existing seeds, I'd recommend doing some kind of scaling on the CFG scale if "CFGv2" is enabled so that 7 roughly equates to 0.15 or whatever is worked out to be the magic number...so like, for 7, you'd divide it by 46.6666repeating.

I basically just thought that that was the equivalent part of this custom code for cfg with the k-diffusion samplers and wanted to see how it would work.

The 2 latter versions of the modifications with the alternate scaling range 0-1 and the "whole" rescaling don't seem to work well for me here, however I got good results from the original cfg scale with "whole" rescaling so I will comment out the latter changes for now on this branch. If anyone has any ideas, much appreciated. Thanks for your question!

hdon96 committed:
Update sd_samplers.py

original cfg with "whole" rescaling seems effective, while the rescaled guidance update seems to not work properly here

AUTOMATIC1111 commented:
this code seems to be experimentation so my preferred way for it to be added to webui is via an extension
